### PR TITLE
Add list of user

### PR DESCRIPTION
**Pull request by ShirshovAlex**
remade add user to internship

### CODE REVIEW

File: src/main/java/by/exadel/internship/controller/InternshipController.java [hunk 1]
```diff
@@ -74,24 +74,24 @@ public UserInternshipDTO saveInternship(@RequestBody UserInternshipDTO internshi
...
     @PutMapping("/{internshipId}")
     @ApiOperation("update internship")
-    public UserInternshipDTO updateInternship(@PathVariable(name = "internshipId") UUID internshipId, @RequestBody UserInternshipDTO internshipDTO){
+    public UserInternshipDTO updateInternship(@PathVariable(name = "internshipId") UUID internshipId, @RequestBody UserInternshipDTO internshipDTO) {
         return internshipService.update(internshipId, internshipDTO);
     }
+
+    @SuperAdminAccessControl
+    @PostMapping("/{internshipId}/addUser")
+    @ApiOperation("Add User to Internship")
```
Comment (OlgaEvdokimova) [inline src/main/java/by/exadel/internship/controller/InternshipController.java]:
```suggestion
    @PostMapping("/{internshipId}/user")
    @ApiOperation("Add User to Internship")
```

File: src/main/java/by/exadel/internship/controller/InternshipController.java [hunk 2]
```diff
@@ -74,24 +74,24 @@ public UserInternshipDTO saveInternship(@RequestBody UserInternshipDTO internshi
...
+    public void addUser(@RequestParam(name = "userId") List<UUID> userId, @PathVariable(name = "internshipId") UUID internshipId) {
+        internshipService.addUser(userId, internshipId);
```
Comment (OlgaEvdokimova) [inline src/main/java/by/exadel/internship/controller/InternshipController.java]: Maybe it's better to use @RequestParam in two cases..?

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/controller/InternshipController.java]: it was before, decided to do like that

File: src/main/java/by/exadel/internship/controller/InternshipController.java (hunk 1 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/controller/InternshipController.java]: idk, i think its better not to change
@kzaretskaya @SergeyKoval @viktarkolbik what do u think?

**Review by OlgaEvdokimova — DISMISSED** [review]

File: src/main/java/by/exadel/internship/controller/InternshipController.java (hunk 1 above)
Comment (SergeyKoval) [reply src/main/java/by/exadel/internship/controller/InternshipController.java]: I don't like both variants. Think about something like:
/{internshipId}/users/assign as long as you actually assign list of users to the internship

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java [hunk 3]
```diff
@@ -133,18 +135,32 @@ public UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO) {
...
-            throw new RuntimeException("User is already assigned to Internship");
+        List<User> assignUsers = internship.getUsers();
+
+        List<User> existingUsers = userRepository.findAllById(userIds);
+
+        List<UUID> existingUsersIds = existingUsers.stream()
+                .map(User::getId)
+                .collect(Collectors.toList());
+
+        if (!userIds.containsAll(existingUsersIds)) {
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java]: Is this check valid? Looks like it should be vice versa

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java [hunk 4]
```diff
@@ -133,18 +135,32 @@ public UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO) {
...
+            throw new NotFoundException("Some Users Not Found");
         }
-        internshipRepository.addUserToInternship(userId, internshipId);
+
+        List<User> alreadyAssigned = assignUsers.stream()
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:156]: Try to refactor stream pipeline in order to skip step with collecting

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java [hunk 5]
```diff
@@ -133,18 +135,32 @@ public UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO) {
...
+                .filter(existingUsers::contains)
+                .collect(Collectors.toList());
+
+        if (!alreadyAssigned.isEmpty()){
+            throw new RuntimeException(alreadyAssigned+" already exists");
+        }
+
+        assignUsers.addAll(existingUsers);
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:164]: Can't get this logic. You are adding ALL EXISTING users...

File: src/main/java/by/exadel/internship/controller/InternshipController.java [hunk 6]
```diff
@@ -74,24 +74,24 @@ public UserInternshipDTO saveInternship(@RequestBody UserInternshipDTO internshi
...
     @ApiOperation("update internship")
-    public UserInternshipDTO updateInternship(@PathVariable(name = "internshipId") UUID internshipId, @RequestBody UserInternshipDTO internshipDTO){
+    public UserInternshipDTO updateInternship(@PathVariable(name = "internshipId") UUID internshipId, @RequestBody UserInternshipDTO internshipDTO) {
         return internshipService.update(internshipId, internshipDTO);
     }
+
+    @SuperAdminAccessControl
+    @PostMapping("/{internshipId}/addUser")
+    @ApiOperation("Add User to Internship")
+    public void addUser(@RequestParam(name = "userId") List<UUID> userId, @PathVariable(name = "internshipId") UUID internshipId) {
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/controller/InternshipController.java]:
```suggestion
    public void addUser(@RequestParam(name = "userIds") List<UUID> userIds, @PathVariable(name = "internshipId") UUID internshipId) {
```

File: src/main/java/by/exadel/internship/service/InternshipService.java [hunk 7]
```diff
@@ -25,7 +25,7 @@
 
     UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO);
 
-    void addUser(UUID userId, UUID internshipId);
+    void addUser(List<UUID> userId, UUID internshipId);
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/service/InternshipService.java]:
```suggestion
    void assignUsers(List<UUID> userIds, UUID internshipId);
```

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java [hunk 8]
```diff
@@ -133,18 +135,32 @@ public UserInternshipDTO saveInternship(UserInternshipDTO internshipDTO) {
...
+    public void addUser(List<UUID> userIds, UUID internshipId) {
+        Internship internship = internshipRepository.findById(internshipId)
                 .orElseThrow(() -> new NotFoundException("Internship Not Found"));
 
-        int exist = internshipRepository.checkUserAssign(userId, internshipId);
-        if (exist > 0) {
-            throw new RuntimeException("User is already assigned to Internship");
+        List<User> assignUsers = internship.getUsers();
+
+        List<User> existingUsers = userRepository.findAllById(userIds);
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:145]: > userRepository.findAllById(userIds)
Please, check hibernate sql  what query is generated (with `IN` or multiple with `=`).

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java (hunk 3 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java]: u right, done

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java (hunk 5 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:164]: EXISTING users - its users which exists in our db but not assign to internship

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:164]: ALL EXISTING - users which we found from received ids

File: src/main/java/by/exadel/internship/controller/InternshipController.java (hunk 6 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/controller/InternshipController.java]: done

File: src/main/java/by/exadel/internship/service/InternshipService.java (hunk 7 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/service/InternshipService.java]: done

File: src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java (hunk 8 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/service/impl/InternshipServiceImpl.java:145]: with IN

**Review by SergeyKoval — DISMISSED** [review]

File: src/main/java/by/exadel/internship/controller/InternshipController.java (hunk 1 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/controller/InternshipController.java]: ok, done

**Review by kzaretskaya — APPROVED** [review]

**Review by OlgaEvdokimova — APPROVED** [review]

**Pull request merged**